### PR TITLE
Added PhotoDoodle Pipeline 

### PR DESCRIPTION
**Pull request by AMEERAZAM08**
# What does this PR do?

Fixes # (issue)

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [x] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [ ] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/diffusers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

@sayakpaul  ,@yiyixuxu and @asomoza

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by AMEERAZAM08** [comment]
@sayakpaul  Please Review this

**Comment by AMEERAZAM08** [comment]
@DN6 please review this PR ,
Thanks

**Comment by AMEERAZAM08** [comment]
@sayakpaul Any other can review this

**Review by a-r-r-o-w — COMMENTED** [review]
Hi, there is an excessive amount of unrelated changes that make this PR hard to review. Could you please rebase with main such that only the changes related to PhotoDoodle persist? You can follow some of the past PRs for an example on the expected changes: #10660, #11428, #10136 

Additionally, I think if we tried shipping this in the community projects folder, we could review it faster

**Comment by AMEERAZAM08** [comment]
I wanted to merge in diffuser
it can easily
` import diffusers from PhotoDoodlePipeline `
@a-r-r-o-w Now Updated .

**Comment by AMEERAZAM08** [comment]
Hi @DN6 , @a-r-r-o-w  change has been done

**Comment by AMEERAZAM08** [comment]
@a-r-r-o-w  can you review this we need it in diffusers pipeline 
cc: @sayakpaul

**Comment by a-r-r-o-w** [comment]
@yiyixuxu Could you take a look here/test if you have the bandwidth?

**Comment by yiyixuxu** [comment]
cc @dg845 can you take a look here?

**Comment by dg845** [comment]
As far as I can tell, this pipeline is essentially identical to `FluxKontextPipeline`, as both accept a conditioning `image` argument which is encoded by the VAE to `image_latents`, packed, and concatenated with the packed `latents` in the patch sequence dimension (dim 1). The example from `pipeline_photodoodle.py` works quite well with `FluxKontextPipeline`:

```python
import torch
from diffusers import FluxKontextPipeline
from diffusers.utils import load_image

pipe = FluxKontextPipeline.from_pretrained("black-forest-labs/FLUX.1-dev", torch_dtype=torch.bfloat16)
pipe = pipe.to("cuda")

pipe.load_lora_weights(
    "nicolaus-huang/PhotoDoodle", adapter_name="pretrain", weight_name="pretrain.safetensors"
)
pipe.load_lora_weights(
    "nicolaus-huang/PhotoDoodle", adapter_name="magiceffects", weight_name="sksmagiceffects.safetensors"
)
pipe.set_adapters(["pretrain", "magiceffects"], [1.0, 1.0])

image = load_image(
    "https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/diffusers/astronaut.jpg"
)
prompt = "add a halo and wings for the astronaut by sksmagiceffects"

output_image = pipe(
    image=image,
    prompt=prompt,
    height=image.height,
    width=image.width,
    num_inference_steps=28,
    guidance_scale=3.5,
    generator=torch.Generator().manual_seed(42),
).images[0]

output_image.save("photodoodle_results.png")
```

photodoodle_results

So I don't think there is a need to merge this PR.

**Comment by AMEERAZAM08** [comment]
I agree that this is present now, but it was not there earlier.

@dg845
Alternatively, we can move the code sample to the community code section.
Please let me know if you would like me to create a pull request for this.

**Comment by AMEERAZAM08** [comment]
@dg845  let me know

**Comment by AMEERAZAM08** [comment]
added in community 
@dg845 
thanks
